ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -168,14 +168,19 @@ type RayClusterConditionType string
 
 // Custom Reason for RayClusterCondition
 const (
+	AllPodRunningAndReady = "AllPodRunningAndReady"
+	HeadPodNotFound       = "HeadPodNotFound"
 	// PodRunningAndReady says that the pod is running and ready.
 	PodRunningAndReady = "PodRunningAndReady"
 	// UnknownReason says that the reason for the condition is unknown.
 	UnknownReason = "Unknown"
 )
 
 const (
-	// HeadPodReady is added in a RayCluster when its Head Pod is ready for requests.
```

Comment:
I changed this comment a little bit because `HeadPodReady` will always be there no matter whether the Head Pod is ready or not; it is just its `status` that will be `true` or `false`.